Tools/paletteable sprite editor/js/lib/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+			return true;
+		}
+		
+		return false;
+	};
+	
+	Thresholding.prototype.getSafeColor = function(x, y, img){
+		
+		if(x >= 0 && x < img.getWidth() && y >= 0 && y < img.getHeight()){
+			return img.getIntComponent0(x, y);
+		}
+		return -1;
+	};
+
+	function ThresholdingNeighborhood(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+
+	ThresholdingNeighborhood.prototype.load = function() {
+		this.setAttribute("neighborhoodSide", 10);
+		this.setAttribute("samplingPixelDistance", 1);
+		this.setAttribute("thresholdPercentageOfAverage", 1.0);
+	}
+
+	ThresholdingNeighborhood.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		var neighborhoodSide = getAttribute("neighborhoodSide");
+		var samplingPixelDistance = getAttribute("samplingPixelDistance");
+		var thresholdPercentageOfAverage = getAttribute("thresholdPercentageOfAverage");
+		
+		for(var y=0; y<imageIn.getHeight(); y++){
+			for(var x=0; x<imageIn.getWidth(); x++){
+				this.theshold(imageIn, imageOut, x, y, thresholdPercentageOfAverage, neighborhoodSide, samplingPixelDistance);
+			}
+		}
+	}
+	
+	ThresholdingNeighborhood.prototype.theshold = function(image, imageOut, x, y, thresholdPercentageOfAverage, side, neighborhoodDistance){
+		
+		var min=-1;
+		var max=-1;
+		var pixels=0;
+		var average=0;
+		
+		var inc = neighborhoodDistance;
+		
+		
+		for(var j=y-(side/2); j<y+(inc+side/2); j+=inc){
+			for(var i=x-(side/2); i<x+(side/2); i+=inc){
+				
+				if(i >= 0 && j>= 0 && i < image.getWidth() && j < image.getHeight()){
+					
+					var color = image.getIntComponent0(i,j);
+					
+					if(min == -1 || color < min){
+						min = color;
+					}
+					if(max == -1 || color > max){
+						max = color;
+					}
+						
+					average+=color;
+					pixels++;
+				}
+			}
+		}
+		
+		average /= pixels;
+			
+		var color = image.getIntComponent0(x,y);
+		
+		if(color < average*thresholdPercentageOfAverage || (max-min) <= 30){
+			imageOut.setIntColor(x, y, 255, 0, 0, 0);
+		} else{
+			imageOut.setIntColor(x, y, 255, 255, 255, 255);
+		}
+	};
+
+	function CombineByAlpha(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+	
+	CombineByAlpha.prototype.load = function(){
+		this.setAttribute("x", 0);
+		this.setAttribute("y", 0);
+		this.setAttribute("imageOther", null);
+	}
+	
+	CombineByAlpha.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attributesOut,
+		mask, 
+		previewMode
+	)
+	{
+		var imageOther = this.getAttribute("imageOther");
+		var x = this.getAttribute("x");
+		var y = this.getAttribute("y");
+		
+		if(imageOther != null){
+			
+			for(var j=0; j<imageIn.getHeight(); j++){
+				for(var i=0; i<imageIn.getWidth(); i++){
+					
+					var ox = i-x;
+					var oy = j-y;
+					
+					if(ox >= 0 && ox < imageOther.getWidth() && oy >= 0 && oy < imageOther.getHeight()){
+						var alpha = imageOther.getAlphaComponent(ox,oy);
+						if(alpha != 0){
```

Comment:
Codacy found an issue: [Use ===/!== to compare with true/false or Numbers](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)